up-l1/README.adoc

```diff
@@ -279,6 +239,23 @@ Access token as defined in per link:../up-l2/permissions.adoc#_token_based_acces
 * *MAY* be of type `String` for Java, or `std::vector<uint8_t>` for C++
 
 
+=== UPayload
+
+UPayload is a container for the uP-L3 application layer data to be transmitted between uEs. The UPayload structure contains only the data, its size, and an optional hint to the serialization format. 
+
+If the data is passed by reference (ex. pointer):
+
+* `data` *MUST* contain the pointer to the payload and `size` contains the actual data size
```

Comment:
need to add another serialization hint type of pointer to make this explicit?  streamer likely needs to know this for data heading out of the device so it can dereference the point to send the actual payload outside the device.  Has to be able to do this without knowledge of the topic.

Reply:
Lets talk about this with the others, it is a valid point